File: longestCommonPrefix.java
class longestCommonPrefix {

    public static void main(String[] args) {
        String[] strs = {"flowers", "flow", "fled"};
        System.out.println(longestCommonPrefix(strs));
    }
    String longestCommonPrefix(String arr[], int n){
        String res = "";
        String minStr = arr[0];
        for (int i = 0; i < arr.length; i++) {
            if (arr[i].length < minStr) {
                minStr = arr[i];
            }
        }
        
        for (int i : arr) {
            for (int j = 0, k = 0; j < minStr.length(); i++) {
                char s = minStr.toCharArray();
            }
        }

        return res;
        
    }

}

File: longestCommonPrefix.py
from typing import List 

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        mini, maxi = [x for x in strs if len(x) == min(len(i) for i in strs)], [x for x in strs if len(x) == max(len(i) for i in strs)]
        mini = mini[0]
        maxi = maxi[0]
        print(mini, maxi, sep = ' ')

        mini, maxi = min(strs), max(strs)
        print(mini, maxi, sep = ' ')

        for i in range(len(mini)):
            if mini[i] != maxi[i]:
                return mini[:i]
        return ""

strs = ["flower","flow","flight"]
print(Solution().longestCommonPrefix(strs))
